controller.xql

```diff
@@ -17,84 +17,93 @@ declare variable $app-root-absolute-url :=
     || $exist:controller
 ;
 
-login:set-user("org.exist.public-repo.login", (), false()),
-
+declare function local:is-authorized-user() as xs:boolean {
+    let $user := request:get-attribute($config:login-domain || ".user")
+    return (
+        exists($user) and 
+        sm:get-user-groups($user) = config:repo-permissions()?group
+    )
+};
+
+login:set-user($config:login-domain, (), false()),
+(: public routes :)
 if ($exist:path eq "") then
     <dispatch xmlns="http://exist.sourceforge.net/NS/exist">
         <redirect url="{request:get-uri()}/"/>
     </dispatch>
 
-else if ($exist:path eq "/") then
+else if (request:get-method() eq "GET" and $exist:path eq "/") then
     (: forward root path to index.xql :)
     <dispatch xmlns="http://exist.sourceforge.net/NS/exist">
-        <redirect url="index.html"/>
-    </dispatch>
-
-else if ($exist:path eq "/public/apps.xml") then
-    <dispatch xmlns="http://exist.sourceforge.net/NS/exist">
-        <forward url="{$exist:controller}/modules/list.xq"/>
-    </dispatch>
-
-(:  Protected resource: user is required to log in with valid credentials.
-    If the login fails or no credentials were provided, the request is redirected
-    to the login.html page. :)
-else if ($exist:path eq "/admin.html") then
-    let $user := request:get-attribute("org.exist.public-repo.login.user")
-    return
-        if (exists($user) and sm:get-user-groups($user) = config:repo-permissions()?group) then
-            <dispatch xmlns="http://exist.sourceforge.net/NS/exist">
-                <view>
-                    <forward url="{$exist:controller}/modules/view.xq">
-                        <set-header name="Cache-Control" value="no-cache"/>
-                    </forward>
-                </view>
-            </dispatch>
-        else
-            <dispatch xmlns="http://exist.sourceforge.net/NS/exist">
-                <forward url="login.html"/>
-                <view>
-                    <forward url="{$exist:controller}/modules/view.xq">
-                        <set-header name="Cache-Control" value="no-cache"/>
-                    </forward>
-                </view>
-            </dispatch>
-
-(:  Protected resource :)
-else if ($exist:path eq "/put-package") then
+        <forward url="/{$exist:controller}/index.html"/>
+        <view>
+            <forward url="/{$exist:controller}/modules/view.xq">
+                <set-header name="Cache-Control" value="no-cache"/>
+            </forward>
+        </view>
+    </dispatch>
+
+(: legacy route :)
+else if (request:get-method() eq "GET" and $exist:path eq "/public/apps.xml") then
     <dispatch xmlns="http://exist.sourceforge.net/NS/exist">
-        <forward url="{$exist:controller}/modules/put-package.xq"/>
+        <forward url="/{$exist:controller}/modules/list.xq"/>
```

Comment:
Here and in the subsequent similar changes, the leading `/` added should be removed. `$exist:controller` already provides a leading `/`. (In this app, it's `/public-repo`.)

Reply:
OK, will do. I found one forward with leading slash and thought it's the way to go.